### PR TITLE
Add restrict-metadata-access

### PR DESCRIPTION
**Pull request by geoffjentry**
One could argue that "auth file" is now a misnomer, but I rationalized my laziness for maintaining the name in that this *relates* to auth. Up to you all.

### CODE REVIEW

**Comment by danbills** [comment]
👍

File: supportedBackends/jes/src/test/scala/cromwell/backend/impl/jes/JesAttributesSpec.scala [hunk 1]
```diff
@@ -46,6 +47,14 @@ class JesAttributesSpec extends FlatSpec with Matchers {
     jesAttributes.computeServiceAccount should be("testing")
   }
 
+  it should "parse restrict-metadata-access" taggedAs IntegrationTest in {
+    val backendConfig = ConfigFactory.parseString(configString(genomics = """restrict-metadata-access = true """))
```
Comment (kshakir) [inline supportedBackends/jes/src/test/scala/cromwell/backend/impl/jes/JesAttributesSpec.scala]: Are the triple quotes necessary?

**Comment by kshakir** [comment]
👍  Minor comments, otherwise LGTM.

File: supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesAttributes.scala [hunk 2]
```diff
@@ -76,6 +78,7 @@ object JesAttributes {
     val maxPollingInterval: Int = backendConfig.as[Option[Int]]("maximum-polling-interval").getOrElse(600)
     val computeServiceAccount: String = backendConfig.as[Option[String]]("genomics.compute-service-account").getOrElse("default")
     val genomicsAuthName: ErrorOr[String] = validate { backendConfig.as[String]("genomics.auth") }
+    val genomicsRestrictMetadataAccess: ErrorOr[Boolean] = validate { backendConfig.as[Option[Boolean]]("genomics.restrict-metadata-access").getOrElse(false) }
```
Comment (mcovarr) [inline supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesAttributes.scala]: should this really be on by default?

**Comment by mcovarr** [comment]
:+1:

File: supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesAttributes.scala (hunk 2 above)
Comment (geoffjentry) [inline supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesAttributes.scala]: discussed in slack

File: supportedBackends/jes/src/test/scala/cromwell/backend/impl/jes/JesAttributesSpec.scala (hunk 1 above)
Comment (geoffjentry) [inline supportedBackends/jes/src/test/scala/cromwell/backend/impl/jes/JesAttributesSpec.scala]: This is either copy/paste or intellij. will look and modify as appropriate.

**Pull request merged**